modules/diskq/qdisk.c

```diff
@@ -537,6 +540,18 @@ qdisk_pop_head(QDisk *self, GString *record)
   return TRUE;
 }
 
+gboolean
+qdisk_pop_head(QDisk *self, GString *record)
+{
+  return _pop_head(self, record);
+}
+
+gboolean
+qdisk_skip_head(QDisk *self)
+{
+  return _pop_head(self, NULL);
```

Comment:
A bool or a null/non-null parameter is usually an indicator that the function has 2 different control flows, which may not result in clean code.

In that case, duplication is better than breaking single responsibility in my opinion.

What do you think about the below alternative?

```C
gboolean
qdisk_pop_head(QDisk *self, GString *record)
{
  guint32 record_length;
  if (_try_reading_record_length(self, &record_length)
    return FALSE;
    
  if (!_read_record_from_disk(self, record, record_length))
    return FALSE;

  _update_positions_after_read(self, record_length); // update_reading_positions()?
  return TRUE;
}

gboolean
qdisk_skip_head(QDisk *self)
{
  guint32 record_length;
  if (_try_reading_record_length(self, &record_length)
    return FALSE;

  _update_positions_after_read(self, record_length);
  return TRUE;
}
```

The only ugly thing here is that `_update_positions_after_read()` will reset the file if it's empty and it's not reliable, but I think this call does not really belong here, so we can move that to a better place in the future.

Reply:
Thanks for the tip, I kept one more function, resulting 4 and 3 in case of pop() and skip(), but I think it is still readable and has not too much copy-paste.